packages/eds-core-react/src/components/Autocomplete/Autocomplete.tsx

```diff
@@ -431,18 +434,21 @@ function AutocompleteInner<T>(
   } = useMultipleSelection(multipleSelectionProps)
 
   const allSelectedState = useMemo(() => {
-    if (!inputOptions || !selectedItems) return 'NONE'
-    if (inputOptions.length === selectedItems.length) return 'ALL'
-    if (inputOptions.length != selectedItems.length && selectedItems.length > 0)
+    if (!notDisabledItems || !selectedItems) return 'NONE'
+    if (notDisabledItems.length === selectedItems.length) return 'ALL'
+    if (
+      notDisabledItems.length != selectedItems.length &&
+      selectedItems.length > 0
+    )
       return 'SOME'
     return 'NONE'
-  }, [inputOptions, selectedItems])
+  }, [notDisabledItems, selectedItems])
 
   const toggleAllSelected = () => {
-    if (selectedItems.length === inputOptions.length) {
+    if (selectedItems.length === notDisabledItems.length) {
       setSelectedItems([])
     } else {
-      setSelectedItems(inputOptions)
+      setSelectedItems(notDisabledItems)
```

Comment:
Sorry for drive-by reviewing and also sorry for potentially adding more work, but:

Consider a situation where one of the disabled options has been selected¹. In this scenario, I would expect the value of the disabled options to be unaffected by toggling select all; i.e. the setSelectedItems needs to take those into account. In addition, the comparison `selectedItems.length === notDisabledItems.length` would be wrong in this scenario.

¹This could e.g. happen in a scenario where users are allowed to apply filters to which end results should be selected, without those filters removing selected values.

Reply:
Ah you are quite right, of course a disabled item can also be selected. So I need a "selectableItems.length" as well basically